common/djangoapps/third_party_auth/strategy.py

```diff
@@ -32,3 +36,44 @@ def setting(self, name, default=None, backend=None):
         # At this point, we know 'name' is not set in a [OAuth2|SAML]ProviderConfig row.
         # It's probably a global Django setting like 'FIELDS_STORED_IN_SESSION':
         return super(ConfigurationModelStrategy, self).setting(name, default, backend)
+
+    def _ensure_passes_length_check(self, user_data, key, fallback, min_length=2):
+        """
+        Ensures that value we get from user_data is meets length requirements. IF it is shorter than required, fallback
+        is used
+        """
+        assert len(fallback) >= min_length
+        value = user_data.get(key)
+        if value and len(value) >= min_length:
+            return value
+        return fallback
+
+    def create_user(self, *args, **kwargs):
+        """
+        # Creates user using information provided by pipeline. This method is called in create_user pipeline step.
+        # Unless the workflow is changed, create_user immediately terminates if the user already found/
+        # So far, user is either created in ensure_user_information via registration form or account needs to be
+        # autoprovisioned. So, this method is only called when autoprovisioning account.
+        """
+        from student.views import create_account_with_params
+        from .pipeline import make_random_password
+
+        user_fields = dict(kwargs)
+        # needs to be >2 chars to pass validation
+        name = self._ensure_passes_length_check(
+            user_fields, 'fullname', self.setting("THIRD_PARTY_AUTH_FALLBACK_FULL_NAME")
+        )
+        password = self._ensure_passes_length_check(user_fields, 'password', make_random_password())
+
+        user_fields['name'] = name
+        user_fields['password'] = password
+        user_fields['honor_code'] = True
+        user_fields['terms_of_service'] = True
+
+        if not user_fields.get('email'):
+            user_fields['email'] = "{username}@{domain}".format(
+                username=user_fields['username'], domain=self.setting("FAKE_EMAIL_DOMAIN")
+            )
+
+        # when autoprovisioning we need to skip email activation, hence skip_email is True
```

Comment:
What's the reason that `autoprovision_account` requires skipping email verification?